jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -588,27 +675,42 @@ ecma_builtin_helper_string_prototype_object_index_of (ecma_string_t *original_st
       {
         break;
       }
-      if (ecma_builtin_helper_string_find_index (original_str_p, search_str_p, true,
-                                                 (lit_utf8_size_t) start_ends_with, &index_of))
+      if (ecma_builtin_helper_string_find_index (original_str_p, search_str_p,
+                                                      (uint32_t) start_ends_with) != UINT32_MAX)
       {
-        ret_value = ecma_make_boolean_value (index_of == (lit_utf8_size_t) start_ends_with);
+        ret_value = ecma_make_boolean_value (
+                                             (ecma_builtin_helper_string_find_index (original_str_p, search_str_p,
```

Comment:
This shouldn't be called twice, you could store the result of the first call in a variable and reuse it.